src/util/transition/ngbTransition.ts

```diff
@@ -0,0 +1,65 @@
+import {EMPTY, fromEvent, Observable, of, race, Subject, timer} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+import {getTransitionDurationMs} from './util';
+import {environment} from '../../environment';
+
+export type NgbTransitionStartFn = (element: HTMLElement) => void;
+
+export interface NgbTransitionOptions {
+  animation: boolean;
+  runningTransition: 'continue' | 'stop';
+}
+
+const TIMER_DELAY_MS = environment.production ? 5 : 500;
```

Comment:
why can't we use 5 everytime ?

Reply:
Will add a comment, but basically some tests become unstable with timeout firing before transitionend

Reply:
Already discussed with @maxokorokov, but I would set the timer delay directly in environment, to avoid the test `environment.production`